storage/innobase/lock/lock0lock.cc

```diff
@@ -2310,8 +2320,8 @@ static void lock_rec_dequeue_from_page(lock_t *in_lock, bool owns_wait_mutex)
 
 	const page_id_t page_id{in_lock->un_member.rec_lock.page_id};
 	auto& lock_hash = lock_sys.hash_get(in_lock->type_mode);
-	ut_ad(lock_sys.is_writer() || in_lock->trx->mutex_is_owner());
-
+	ut_ad(lock_sys.is_writer() || in_lock->trx->mutex_is_owner()
+		|| lock_sys.is_cell_locked(*in_lock));
```

Comment:
In the old InnoDB formatting style, which this function is following, lines should be indeed split before binary operators such as `||`. But, the continuation lines should be indented so that they align with the preceding `(`. In this case, instead of TAB+TAB the indentation should be TAB+6 spaces.

There are a few other places like this in this file.